packages/@glimmerx/component/ember-addon-main.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+function makePrecompileTemplate(templateCompiler) {
+  const precompileTemplate = (templateString, templateTokens, options) => {
+    let compiled = templateCompiler.precompile(templateString, options);
+
+    return `Ember.HTMLBars.template(${compiled})`;
+  };
```

Comment:
Can we leverage ember-cli-htmlbars here? We know the host will already have it...

Reply:
How do you mean exactly? Instead of loading the template compiler here?

Reply:
Yeah, the template compiler loading is "tricky". The current implementation will not support the app having AST tranforms (e.g. for `{{t`/`{{t-def`, et al) because it doesn't load plugins.

This is fine for now, but I'd like to add an API to ember-cli-htmlbars that we can just use directly.